operator/pkg/ciliumidentity/controller_test.go

```diff
@@ -197,3 +204,253 @@ func verifyCIDUsageInCES(ctx context.Context, fakeClient *k8sClient.FakeClientse
+		expectedInCIDStore [][]map[string]string
+	}{
+		{
+			name: "create_cid",
+			podActions: []podAction{
+				{"create", pod1, nil},
+			},
+			expectedInCIDStore: [][]map[string]string{
+				{cid1.SecurityLabels},
+			},
+		},
+		{
+			name: "create_two_cids",
+			podActions: []podAction{
+				{"create", pod1, nil},
+				{"create", pod2, nil},
+			},
+			expectedInCIDStore: [][]map[string]string{
+				{cid1.SecurityLabels},
+				{cid1.SecurityLabels, cid2.SecurityLabels},
+			},
+		},
+		{
+			name: "create_and_delete_cid",
+			podActions: []podAction{
+				{"create", pod1, nil},
+				{"delete", pod1, nil},
+			},
+			expectedInCIDStore: [][]map[string]string{
+				{cid1.SecurityLabels},
+				// TODO: Update the test to actually delete CID after the pod is deleted.
+				// https://github.com/cilium/cilium/issues/34740
+				{cid1.SecurityLabels},
+			},
+		},
+		{
+			name: "recreate_cid",
+			podActions: []podAction{
+				{"create", pod1, nil},
+				{"delete", pod1, nil},
+				{"create", pod1, nil},
+			},
+			expectedInCIDStore: [][]map[string]string{
+				{cid1.SecurityLabels},
+				// TODO: Update the test to actually delete CID after the pod is deleted.
+				// https://github.com/cilium/cilium/issues/34740
+				{cid1.SecurityLabels},
+				{cid1.SecurityLabels},
+			},
+		},
+		{
+			name: "recreate_cid_on_pod_update",
+			podActions: []podAction{
+				{"create", pod1, nil},
+				{"update", updatedPod, nil},
+			},
+			expectedInCIDStore: [][]map[string]string{
+				{cid1.SecurityLabels},
+				// TODO: Update the test to actually delete CID after the pod is deleted.
+				// https://github.com/cilium/cilium/issues/34740
+				{cid1.SecurityLabels, cid2.SecurityLabels},
+			},
+		},
+		{
+			name: "reuse_cid",
+			podActions: []podAction{
+				{"create", pod1, nil},
+				{"delete", pod1, nil},
+				{"create", pod1, nil},
+			},
+			expectedInCIDStore: [][]map[string]string{
+				{cid1.SecurityLabels},
+				{cid1.SecurityLabels},
+				{cid1.SecurityLabels},
+			},
+		},
+		{
+			name: "update_cid",
+			podActions: []podAction{
+				{"create", pod1, nil},
+				{},
+			},
+			cidUpdateList: []*capi_v2.CiliumIdentity{
+				nil,
+				cid2,
+			},
+			expectedInCIDStore: [][]map[string]string{
+				{cid1.SecurityLabels},
+				{cid2.SecurityLabels},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			cid, _, ns, fakeClient, _, h := initHiveTest(true)
+			ctx := context.Background()
+			tlog := hivetest.Logger(t)
+			if err := h.Start(tlog, ctx); err != nil {
+				t.Fatalf("starting hive encountered an error: %s", err)
+			}
+			defer func() {
+				if err := h.Stop(tlog, ctx); err != nil {
+					t.Fatalf("stopping hive encountered an error: %v", err)
+				}
+			}()
+
+			nsStore, err := (*ns).Store(ctx)
+			if err != nil {
+				t.Fatalf("get namespace store: %v", err)
+			}
+			if err := nsStore.CacheStore().Add(ns1); err != nil {
+				t.Fatalf("create namespace: %v", err)
+			}
```

Comment:
Any particular reason for injecting the namespace here instead of creating through clientset?

Reply:
You're right, clientset can be used. I reverted the namespace store changes.